site/content/docs/development/tutorials/runnable.md

```diff
@@ -0,0 +1,528 @@
+complete the scenario by using Runnable in a supply chain.
+
+## Steps
+
+Before using Cartographer, let’s think about how we would use Tekton on its
+own to lint a repo. First we would define a pipeline:
+
+```yaml
+---
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: linter-pipeline
+spec:
+  params:
+    - name: repository
+      type: string
+    - name: revision
+      type: string
+  workspaces:
+    - name: shared-workspace
+  tasks:
+    - name: fetch-repository
+      taskRef:
+        name: git-clone
+      workspaces:
+        - name: output
+          workspace: shared-workspace
+      params:
+        - name: url
+          value: $(params.repository)
+        - name: revision
+          value: $(params.revision)
+        - name: subdirectory
+          value: ""
+        - name: deleteExisting
+          value: "true"
+    - name: md-lint-run
+      taskRef:
+        name: markdown-lint
+      runAfter:
+        - fetch-repository
+      workspaces:
+        - name: shared-workspace
+          workspace: shared-workspace
+      params:
+        - name: args
+          value: ["."]
+```
+
+We would apply this pipeline to the cluster, along with the tasks.
+Those tasks are in the TektonCD Catalog:
+
+```shell
+$ kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/main/task/git-clone/0.3/git-clone.yaml
+$ kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/main/task/markdown-lint/0.1/markdown-lint.yaml
+```
+
+Finally we need to create a pipeline-run object. This object provides the
+param and workspace values defined at the top level `.spec` field of the
+pipeline.
+
+```yaml
+---
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: linter-pipeline-run
+spec:
+  pipelineRef:
+    name: linter-pipeline
+  params:
+    - name: repository
+      value: https://github.com/waciumawanjohi/demo-hello-world
+    - name: revision
+      value: main
+  workspaces:
+    - name: shared-workspace
+      volumeClaimTemplate:
+        spec:
+          accessModes:
+            - ReadWriteOnce
+          resources:
+            requests:
+              storage: 256Mi
+```
+
+Importantly, this pipeline-run object will kick off a single run
+of the pipeline. The run will either succeed or fail. The outcome will be
+written into the pipeline-run’s status. No later changes to the pipeline-run
+object will change those outcomes; the run happens once.
+
+To see this in action, let’s apply the above pipeline-run. If we watch the
+object, we’ll soon see that it succeeds.
+
+```shell
+$ watch 'kubectl get -o yaml pipelinerun linter-pipeline-run | yq .status.conditions'
+```
+
+Eventually yields the result:
+
+```json
+[
+  {
+    "lastTransitionTime": ...,
+    "message": "Tasks Completed: 2 (Failed: 0, Cancelled 0), Skipped: 0",
+    "reason": "Succeeded",
+    "status": "True",
+    "type": "Succeeded"
+  }
+]
+```
```

Comment:
this command is using the non-standard version of `yq`. the more standard version (which more people will likely be using) will produce this output:
```suggestion
```yaml
- lastTransitionTime: ...
  message: "Tasks Completed: 2 (Failed: 0, Cancelled 0), Skipped: 0"
  reason: "Succeeded"
  status: "True"
  type: "Succeeded"
```
```